collector.go

```diff
@@ -79,13 +80,21 @@ type nextcloudCollector struct {
 	scrapeErrorsMetric prometheus.Counter
 }
 
-func newCollector(infoURL, username, password string, timeout time.Duration, userAgent string) *nextcloudCollector {
+func newCollector(infoURL, username, password string, timeout time.Duration, userAgent string, insecure bool) *nextcloudCollector {
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{
+			// disable TLS certification verification, if desired
+			InsecureSkipVerify: insecure,
+		},
+	}
+
 	return &nextcloudCollector{
 		infoURL:  infoURL,
 		username: username,
 		password: password,
 		client: &http.Client{
-			Timeout: timeout,
+			Timeout:   timeout,
+			Transport: tr,
```

Comment:
I'd say `tr` can be inlined into the struct definition.

Reply:
Sure thing!